doc: terminology cleanup in HLD overview (#6754)
projectacrn/acrn-hypervisor: doc/developer-guides/hld/hld-overview.rst

ACRN Architecture
*****************

ACRN is a type 1 hypervisor that runs on top of bare metal. It supports
certain :ref:`Intel platforms <hardware>` and can be easily extended to support
future
platforms. ACRN implements a hybrid VMM architecture, using a privileged
Service VM to manage I/O devices and
provide I/O mediation. Multiple User VMs can be supported, running Ubuntu
or Android OS.

ACRN 1.0
========

ACRN 1.0 is designed mainly for auto use cases such as SDC and IVI.

> Instrument cluster applications are critical in the Software Defined

Reviewer: Can we drop the definitions of the acronyms here?

ACRN 2.0
========

> ACRN 2.0 is extending ACRN to support a pre-launched VM (mainly for safety VM)

Reviewer: ```suggestion
ACRN 2.0 extended ACRN to support a pre-launched VM (mainly for safety VM)
```

and real-time (RT) VM.

:numref:`overview-arch2.0` shows the architecture of ACRN 2.0; the main
differences compared to ACRN 1.0 are that:

-  ACRN 2.0 supports a pre-launched VM, with isolated resources,
   including CPU, memory, and hardware devices.

-  ACRN 2.0 adds a few necessary device emulations in the hypervisor, such as
   vPCI and vUART, to avoid interference between different VMs.

> -  ACRN 2.0 supports an RT VM as a post-launched User VM, with features such as

Reviewer: I don't think I saw a definition of RT yet, or I missed it.

Author: It's in line 131.

   LAPIC passthrough and PMD virtio driver.

> ACRN 2.0 is still WIP, and some of its features are already merged in the master.

Author: @dbkinder Should I remove line 146? ("ACRN 2.0 is still WIP ... ")

Reviewer: Indeed, since 2.0 was release in Jun 2020, please remove this WIP comment

ACRN takes advantage of Intel Virtualization Technology (Intel VT).
The ACRN HV runs in Virtual Machine Extension (VMX) root operation,
host mode, or VMM mode, while the Service VM and User VM guests run
in VMX non-root operation, or guest mode. (We'll use "root mode"
and "non-root mode" for simplicity.)

The VMM mode has 4 rings. ACRN
runs the HV in ring 0 privilege only, and leaves ring 1-3 unused. A guest
running in non-root mode has its own full rings (ring 0 to 3). The

> guest kernel runs in ring 0 in guest mode, while the guest user land

Reviewer: I think we're using "userland" (one word)

VM Manager creates the User VM based on DM application, and does User VM state
management by interacting with lifecycle service in ACRN service.

> Refer to VM management chapter for more details. <link?>

Author: Is there a VM Manager chapter?

Reviewer: Yes, ``:ref:`hv-vm-management` ``

functionalities. HSM communicates with the ACRN hypervisor through
hypercall or upcall interrupts.

Refer to the HSM chapter for more details. <link?>

> 

Author: Is there an HSM chapter?

Reviewer: Here's a link to the section (https://projectacrn.github.io/latest/developer-guides/hld/hld-devicemodel.html#hsm) but you'll need to add a label in hld-devicemodel.rst at that point so you can add a direct link to it here.

Open Virtual Machine Firmware (OVMF) is the virtual bootloader that supports
the EFI boot of the User VM on the ACRN hypervisor platform.

The OVMF is copied to the User VM memory by the VM Manager in the Service VM

> while creating the User VM virtual BSP of the User VM. The Service VM passes the

Reviewer 2: "while creating" --> Because of the passive voice ("is copied by"), the grammatical subject of the sentence is _The OVMF_, so technically "while creating" refers to the OVMF (which I'm betting isn't correct). So a touch of rewriting is required here.

-  Mitigation of DMA storm.

>    (To be documented later.) <Remove?>

Reviewer: @NanlinXie Can we get engineering to provide any details about how ACRN mitigates a DMA storm?  If not, we should delete this bullet point.

Reviewer: I'd just recommend we delete the bullet point and line 528 for now.

=====================

In ACRN, CPU P-state and C-state (Px/Cx) are controlled by the guest OS.

> The corresponding governors are managed in the Service VM/User VM for best power

Reviewer: ```suggestion
The corresponding governors are managed in the Service VM or User VM for best power
```

efficiency and simplicity.

Guests should be able to process the ACPI P/C-state request from OSPM.
The needed ACPI objects for P/C-state management should be ready in an
ACPI table.

> The hypervisor can restrict a guest's P/C-state request (per customer

Reviewer: ```suggestion
The hypervisor can restrict a guest's P-state and C-state request (per customer
```

requirement). MSR accesses of P-state requests could be intercepted by
the hypervisor and forwarded to the host directly if the requested

> P-state is valid. Guest MWAIT/Port IO accesses of C-state control could

Reviewer: ```suggestion
P-state is valid. Guest MWAIT or Port IO accesses of C-state control could
```

   :align: center

>    CPU P/C-state Management Block Diagram

Reviewer: ```suggestion
   CPU P-state and C-state Management Block Diagram
```
